hack/derive-version.sh

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+# Copyright The Enterprise Contract Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Assumptions:
+# 1. The file VERSION exists and contains the major.minor version number
+# 2. The most recent commit that changed that file is the commit that
+#    bumped the minor version number.
+# 3. The only way that a change arrives in main branch is via a merge
```

Comment:
Sometimes we do that. Maybe before merging this, we should restrict pushes?

Reply:
Yeah, that might be a good idea either way, right? Pushing directly bypasses pre-merge CI, right? Is there are reasonable scenario where we do want to be able to push direct to main?

Reply:
Looks like this is unchecked "Do not allow bypassing the above settings", so we have the rule about requiring a PR but we allow admins to ignore it.

Reply:
Maybe we can also disallow admins to bypass it. If we really need it, we should have enough access to re-allow it.